packages/json/src/index.js

```diff
@@ -11,15 +11,21 @@ export default function json(options = {}) {
     transform(json, id) {
       if (id.slice(-5) !== '.json' || !filter(id)) return null;
 
-      return {
-        code: dataToEsm(JSON.parse(json), {
-          preferConst: options.preferConst,
-          compact: options.compact,
-          namedExports: options.namedExports,
-          indent
-        }),
-        map: { mappings: '' }
-      };
+      try {
+        return {
+          code: dataToEsm(JSON.parse(json), {
+            preferConst: options.preferConst,
+            compact: options.compact,
+            namedExports: options.namedExports,
+            indent
+          }),
+          map: { mappings: '' }
+        };
+      } catch (err) {
+        const error = new Error(`Could not parse ${id}`);
+        error.originalError = err;
+        throw error;
```

Comment:
I sat on this for a while and I'm glad that I did. I'd like to see this use [`this.warn`](https://www.rollupjs.org/guide/en/#thiswarnwarning-string--rollupwarning-position-number---column-number-line-number---void) rather than throwing the error.

Reply:
updated